WcaOnRails/app/models/registration.rb

```diff
@@ -8,9 +8,9 @@ class Registration < ApplicationRecord
   scope :with_payments, -> { joins(:registration_payments).distinct }
 
   belongs_to :competition
-  belongs_to :user
-  belongs_to :accepted_user, foreign_key: "accepted_by", class_name: "User"
-  belongs_to :deleted_user, foreign_key: "deleted_by", class_name: "User"
+  belongs_to :user, optional: true # A user may be deleted later. We only enforce validation directly on creation further down below.
```

Comment:
Can it? :o

As far as I remember we've always had a policy to never destroy users (but instead anonymize them if needed).
I wouldn't be surprised if some parts of the code just assume the user will always be there.

Reply:
We have a test which suggests that it is possible to have a reg without user:
https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/spec/models/registration_spec.rb#L24

Reply:
Wow that is weird :s

From the dev dump:
```
irb(main):001:0> Registration.where(user_id: nil).size
   (349.1ms)  SELECT COUNT(*) FROM `registrations` WHERE `registrations`.`user_id` IS NULL
=> 0 
```

Just pointing out a single use of user which would just break the whole registrations page if it wasn't there: https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/app/views/registrations/edit_registrations.html.erb#L109 (there are probably others).

I'd argue we definitely want it there, but it was introduced by #4799, which makes me think there is more than we can see (and that running the same query on prod would not return 0 :p).
We should probably check what happens when a user is banned (which is hopefully not too often), but I'd bet a reasonable amount of money that something would break ^^

Reply:
I'll be the first in line to remove that weird "verify existence only on create" check.
For the sake of this PR, I only considered the "status quo" and I didn't bother digging into the historic details of this design decision.

Reply:
With warm regards from our production machine:
![image](https://user-images.githubusercontent.com/6136469/188839830-78d77035-aa4f-4804-87ac-5f3a91291e54.png)